### ISSUE TITLE
Feature Request - Add support for resource specific destination table in diagnostic setting policy for API Management

### ISSUE THREAD

**Issue by chrholt** [Azure/Enterprise-Scale#1307]
We would like to have the policy definition for setting diagnostic setting on API Management to support "resource specific" destination table. I suggest adding a parameter for `logAnalyticsDestinationType` in the policy definition which defaults to `AzureDiagnostics`, but can be set to `Dedicated`.

Add the following parameter in the [Deploy Diagnostic Settings for API Management to Log Analytics workspace](https://github.com/Azure/Enterprise-Scale/blob/main/src/resources/Microsoft.Authorization/policyDefinitions/Deploy-Diagnostics-APIMgmt.json) policy definition's parameters.
```
"logAnalyticsDestinationType": {
        "type": "String",
        "defaultValue": "AzureDiagnostics",
        "allowedValues": [
          "AzureDiagnostics",
          "Dedicated"
        ],
        "metadata": {
          "displayName": "Log Analytics Destination Type",
          "description": "Set the destination table for the diagnostic setting in Log Analytics workspace - AzureDiagnostics or Dedicated (resource specific)"
        }
      }
```

And also use this parameter in the `logAnalyticsDestinationType` property in the deployment template.

**Comment by Springstone** [comment]
Hi @chrholt, thank you for raising this feature request. I've confirmed the API/RP support the requested parameter, just need to do some validation on allowed values, as documentation indicates null and Dedicated as allowed. 
If there are no issues, we'll add this to our backlog as it is pretty straight forward to implement.

**Comment by chrholt** [comment]
Wonderful. I've already implemented this as a custom policy on our end so I can create a pull request if you'd like?

**Comment by Springstone** [comment]
Of course, we welcome contributions to ALZ. Please do follow https://github.com/Azure/Enterprise-Scale/wiki/ALZ-Contribution-Guide as always :) I see you have contributed in the past, so this should be clear :)

**Comment by picccard** [comment]
Commenting here for ref, as similar requests are beeing captured in #1093

**Comment by Springstone** [comment]
This has been resolved in #1324
